doc/en/components/grids/_shared/column-resizing.md

```diff
@@ -0,0 +1,538 @@
+
+You can also auto-size a column dynamically using the exposed `Autosize` method on `ColumnComponent`.
+
+<!-- ComponentStart: Grid, TreeGrid -->
+
+```typescript
+@ViewChild('@@igObjectRef') @@igObjectRef: {ComponentName};
+
+let column = this.@@igObjectRef.columnList.filter(c => c.field === 'ID')[0];
+column.autosize();
+```
+
+```razor
+@code {
+    private {ComponentName} gridRef;
+
+    protected void OnInitialize()
+    {
+        IgbGridColumn column = gridRef.Columns.Where((col) => { return col.Field == "ID"; }).FirstOrDefault();
+        column.Autosize(false);
+    }
+}
+```
+
+<!-- ComponentEnd: Grid, TreeGrid -->
+<!-- ComponentStart: HierarchicalGrid -->
+
+```typescript
+@ViewChild('@@igObjectRef') @@igObjectRef: {ComponentName};
+
+let column = this.@@igObjectRef.columnList.filter(c => c.field === 'Artist')[0];
+column.autosize();
+```
+
+```razor
+@code {
+    private {ComponentName} gridRef;
+
+    protected void OnInitialize()
+    {
+        IgbGridColumn column = gridRef.Columns.Where((col) => { return col.Field == "Artist"; }).FirstOrDefault();
+        column.Autosize(false);
+    }
+}
+```
+
+<!-- ComponentEnd: HierarchicalGrid -->
+
+## Auto-size columns on initialization
+
+Each column can be set to auto-size on initialization by setting `Width` to 'auto':
+
+```html
+<igx-column width='auto'>...
+```
+
+```razor
+<IgbGridColumn Width="auto"></IgbGridColumn>
+```
+
+When the column is first initialized in the view it resolves its width to the size of the longest visible cell or header. Note that cells that are outside of the visible rows are not included.
+This approach is more performance optimized than auto-sizing post initialization and is recommended especially in cases where you need to auto-size a large number of columns.
+
+<!-- ComponentStart: Grid -->
+
+<code-view style="height:550px" 
+           data-demos-base-url="{environment:demosBaseUrl}" 
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-columnAutoResizing-sample" alt="{Platform} {ComponentTitle} Column Resizing Example">
+</code-view>
+
+<!-- ComponentEnd: Grid -->
+<!-- ComponentStart: TreeGrid -->
+
+<code-view style="height:550px" 
+           data-demos-base-url="{environment:demosBaseUrl}" 
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-columnAutoSizing-sample" alt="{Platform} {ComponentTitle} Column Resizing Example">
+</code-view>
+
+<!-- ComponentEnd: TreeGrid -->
+<!-- ComponentStart: HierarchicalGrid -->
+
+<code-view style="height:510px" 
+           data-demos-base-url="{environment:demosBaseUrl}" 
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-columnAutosizing-sample" alt="{Platform} {ComponentTitle} Column Resizing Example">
+</code-view>
```

Comment:
remove 2 out of 3 code-view blocks and Component build flags because will use the same routing path for Grid, TreeGrid, HierarchicalGrid in samples.